<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TrimFaceDialog</class>
 <widget class="QWidget" name="TrimFaceDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>320</width>
    <height>480</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Trim Face</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>8</number>
   </property>
   <property name="leftMargin">
    <number>10</number>
   </property>
   <property name="topMargin">
    <number>10</number>
   </property>
   <property name="rightMargin">
    <number>10</number>
   </property>
   <property name="bottomMargin">
    <number>10</number>
   </property>
   <item>
    <widget class="QLabel" name="titleLabel">
     <property name="text">
      <string>Trim Face with Curve</string>
     </property>
     <property name="styleSheet">
      <string notr="true">font-weight: bold; font-size: 11pt; padding: 4px;</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="statusLabel">
     <property name="text">
      <string>Select trimming edges...</string>
     </property>
     <property name="styleSheet">
      <string notr="true">color: #0066cc; font-style: italic; padding: 4px; background-color: #f0f0f0; border-radius: 3px;</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="curvesGroup">
     <property name="title">
      <string>Trimming Edges</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="curveHintLabel">
        <property name="text">
         <string>Click edges in 3D view (Ctrl/Shift for multiple)</string>
        </property>
        <property name="styleSheet">
         <string notr="true">color: #666; font-size: 9pt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QListWidget" name="curveListWidget">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>90</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="curveButtonLayout">
        <item>
         <widget class="QPushButton" name="clearCurvesButton">
          <property name="text">
           <string>Clear All</string>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>26</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="removeCurveButton">
          <property name="text">
           <string>Remove Selected</string>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>26</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="extensionGroup">
     <property name="title">
      <string>Curve Extension</string>
     </property>
     <property name="visible">
      <bool>false</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_extension">
      <item>
       <widget class="QLabel" name="extensionWarningLabel">
        <property name="text">
         <string>[!] Curve appears shorter than surface</string>
        </property>
        <property name="styleSheet">
         <string notr="true">color: #856404; background-color: #fff3cd; border: 1px solid #ffeaa7; border-radius: 3px; padding: 6px; font-weight: bold;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="extensionHintLabel">
        <property name="text">
         <string>Choose how to handle curve extension:</string>
        </property>
        <property name="styleSheet">
         <string notr="true">color: #666; font-size: 9pt; padding-top: 4px;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="extensionNoneRadio">
        <property name="text">
         <string>No extension (use as-is)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="extensionBoundaryRadio">
        <property name="text">
         <string>Extend to surface boundaries</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="extensionCustomRadio">
        <property name="text">
         <string>Custom extension:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="customExtensionWidget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_extension">
         <property name="leftMargin">
          <number>20</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLineEdit" name="extensionDistanceEdit">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="maximumSize">
            <size>
             <width>80</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="placeholderText">
            <string>10.0</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="extensionUnitLabel">
           <property name="text">
            <string>mm beyond surface</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_extension">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="faceGroup">
     <property name="title">
      <string>Face to Trim</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QLabel" name="faceLabel">
        <property name="text">
         <string>No face selected</string>
        </property>
        <property name="styleSheet">
         <string notr="true">padding: 4px; background-color: #fafafa; border: 1px solid #ddd; border-radius: 3px;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="clearFaceButton">
        <property name="text">
         <string>Clear Face</string>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>26</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="pointGroup">
     <property name="title">
      <string>Point to Keep</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QLabel" name="pointLabel">
        <property name="text">
         <string>No point selected</string>
        </property>
        <property name="styleSheet">
         <string notr="true">padding: 4px; background-color: #fafafa; border: 1px solid #ddd; border-radius: 3px;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="clearPointButton">
        <property name="text">
         <string>Clear Point</string>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>26</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="directionGroup">
     <property name="title">
      <string>Projection Direction</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QLabel" name="directionHintLabel">
        <property name="text">
         <string>Choose how the trimming curve projects onto the face:</string>
        </property>
        <property name="styleSheet">
         <string notr="true">color: #666; font-size: 9pt;</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="directionNormalRadio">
        <property name="text">
         <string>Face Normal (perpendicular to surface)</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="directionViewRadio">
        <property name="text">
         <string>View Direction (current camera angle)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="directionCustomRadio">
        <property name="text">
         <string>Custom Vector</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="customVectorWidget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="leftMargin">
          <number>20</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="vectorLabel">
           <property name="text">
            <string>Vector:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="vectorXEdit">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="placeholderText">
            <string>X</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="vectorYEdit">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="placeholderText">
            <string>Y</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="vectorZEdit">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="placeholderText">
            <string>Z</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="buttonLayout">
     <item>
      <widget class="QPushButton" name="applyButton">
       <property name="text">
        <string>Apply</string>
       </property>
       <property name="minimumSize">
        <size>
         <width>90</width>
         <height>32</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton { font-weight: bold; }</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cancelButton">
       <property name="text">
        <string>Cancel</string>
       </property>
       <property name="minimumSize">
        <size>
         <width>90</width>
         <height>32</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
